File: MiniDOMFinanziaria/Finanziaria/StudiDiFattibilita/CQSPDSoluzioniXStudioDiFattibilita.cs
using System;

namespace minidom
{
    public partial class Finanziaria
    {
        [Serializable]
        public class CQSPDSoluzioniXStudioDiFattibilita : CCollection<CQSPDConsulenza>
        {
            private CQSPDStudioDiFattibilita m_StudioDiFattibilita;

            public CQSPDSoluzioniXStudioDiFattibilita()
            {
                m_StudioDiFattibilita = null;
            }

            public CQSPDSoluzioniXStudioDiFattibilita(CQSPDStudioDiFattibilita gruppo) : this()
            {
                Load(gruppo);
            }

            protected override void OnInsert(int index, object value)
            {
                if (m_StudioDiFattibilita is object)
                    ((CQSPDConsulenza)value).SetStudioDiFattibilita(m_StudioDiFattibilita);
                base.OnInsert(index, value);
            }

            protected override void OnSet(int index, object oldValue, object newValue)
            {
                if (m_StudioDiFattibilita is object)
                    ((CQSPDConsulenza)newValue).SetStudioDiFattibilita(m_StudioDiFattibilita);
                base.OnSet(index, oldValue, newValue);
            }

            protected internal void SetStudioDiFattibilita(CQSPDStudioDiFattibilita value)
            {
                if (value is null)
                    throw new ArgumentNullException("value");
                m_StudioDiFattibilita = value;
                for (int i = 0, loopTo = Count - 1; i <= loopTo; i++)
                {
                    var c = this[i];
                    c.SetStudioDiFattibilita(value);
                }
            }

            private void Load(CQSPDStudioDiFattibilita studio)
            {
                if (studio is null)
                    throw new ArgumentNullException("gruppo");
                Clear();
                m_StudioDiFattibilita = studio;
                if (DBUtils.GetID(studio) == 0)
                    return;
                var cursor = new CQSPDConsulenzaCursor();
                try
                {
                    cursor.IDStudioDiFattibilita.Value = DBUtils.GetID(studio);
                    cursor.Stato.Value = ObjectStatus.OBJECT_VALID;
                    cursor.IgnoreRights = true;
                    // cursor.ID.SortOrder = SortEnum.SORT_DESC
                    while (!cursor.EOF())
                    {
                        Add(cursor.Item);
                        cursor.MoveNext();
                    }
                }
                catch (Exception ex)
                {
                    throw;
                }
                finally
                {
                    cursor.Dispose();
                }

                Sort();
            }
        }
    }
}

File: MiniDOMFinanziaria/Finanziaria/StudiDiFattibilita/CQSPDSoluzioniXStudioDiFattibilita.vb
Imports minidom
Imports minidom.Databases
Imports minidom.Sistema
Imports minidom.Anagrafica



Partial Public Class Finanziaria



    <Serializable>
    Public Class CQSPDSoluzioniXStudioDiFattibilita
        Inherits CCollection(Of CQSPDConsulenza)

        Private m_StudioDiFattibilita As CQSPDStudioDiFattibilita

        Public Sub New()
            Me.m_StudioDiFattibilita = Nothing
        End Sub

        Public Sub New(ByVal gruppo As CQSPDStudioDiFattibilita)
            Me.New()
            Me.Load(gruppo)
        End Sub

        Protected Overrides Sub OnInsert(index As Integer, value As Object)
            If (Me.m_StudioDiFattibilita IsNot Nothing) Then DirectCast(value, CQSPDConsulenza).SetStudioDiFattibilita(Me.m_StudioDiFattibilita)
            MyBase.OnInsert(index, value)
        End Sub

        Protected Overrides Sub OnSet(index As Integer, oldValue As Object, newValue As Object)
            If (Me.m_StudioDiFattibilita IsNot Nothing) Then DirectCast(newValue, CQSPDConsulenza).SetStudioDiFattibilita(Me.m_StudioDiFattibilita)
            MyBase.OnSet(index, oldValue, newValue)
        End Sub

        Protected Friend Sub SetStudioDiFattibilita(ByVal value As CQSPDStudioDiFattibilita)
            If (value Is Nothing) Then Throw New ArgumentNullException("value")
            Me.m_StudioDiFattibilita = value
            For i As Integer = 0 To Me.Count - 1
                Dim c As CQSPDConsulenza = Me(i)
                c.SetStudioDiFattibilita(value)
            Next
        End Sub

        Private Sub Load(ByVal studio As CQSPDStudioDiFattibilita)
            If (studio Is Nothing) Then Throw New ArgumentNullException("gruppo")
            Me.Clear()
            Me.m_StudioDiFattibilita = studio
            If (GetID(studio) = 0) Then Exit Sub
            Dim cursor As New CQSPDConsulenzaCursor
            Try
                cursor.IDStudioDiFattibilita.Value = GetID(studio)
                cursor.Stato.Value = ObjectStatus.OBJECT_VALID
                cursor.IgnoreRights = True
                'cursor.ID.SortOrder = SortEnum.SORT_DESC
                While Not cursor.EOF
                    Me.Add(cursor.Item)
                    cursor.MoveNext()
                End While
            Catch ex As Exception
                Throw
            Finally
                cursor.Dispose()
            End Try

            Me.Sort()
        End Sub
    End Class

End Class
